test/test_c10d.py

```diff
@@ -440,7 +440,7 @@ def is_master(self):
         return self.rank == 0
 
 
-class ProcessGroupGlooTest(MultiProcessTestCase):
+class ProcessGroupMultiProcessTest(MultiProcessTestCase):
```

Comment:
I like to run just this one specifically and only run Gloo backend tests (`python test/test_c10d.py ProcessGroupGlooTest`). Instead of overloading this one with both NCCL and Gloo tests, I propose moving the stress functions outside the class, and call them from ProcessGroupGlooTest and ProcessGroupNCCLTest respectively.

That the Gloo backend and NCCL backend tests use multi process and single process respectively is a side effect.

Reply:
A little hesitated to land this test now since CI is broken, possible the old GPUs.